### PR TITLE
Remove run ID and introduce deploy ID

### PR DESCRIPTION
**Pull request by gmega**
This PR removes the notion of a run id and replaces it with a deploy id in continuous tests. Deploy ids can be set at deploy time only (duh), and will be picked up by the test runner from the deploy file on subsequent runs of the continuous test runner. As a consequence, the deploy id becomes a deployer parameter, and can no longer be overridden at the runner. For non-continuous tests, the deploy ID is created on-the-fly.

### CODE REVIEW

**Comment by veaceslavdoina** [comment]
So, there no `runid` env var anymore?

- will `deployment id` be generated automatically at deployment?
- can we still set out own `deployment id` via env var?
- what is the way to get same unique value through the run for
  - Prometheus - populated from Pod labels
  - Elasticsearch - populated from _STATUS.log

I did a test run via GitHub Actions and there we still assign `runid` label/env manually via template, but
- I can't see `deploymentid` label - this is used by Prometheus
- In `09-56-44Z_ContinuousTestRun_STATUS.log` I see `"runid":"20240206-095227"`
- Actual `runid` set by GitHub Actions was `20240206-095044` - I will remove that when we will switch to `deployment id`

**Comment by gmega** [comment]
> So, there no `runid` env var anymore?

Correct, now there's only a deploy id which you can set you deploy a new network.

>     * will `deployment id` be generated automatically at deployment?

Yes, if nothing is set. You can still customize it with `DEPLOYID` when doing the deploy, but you can no longer set it when you fire the test runner, as that's the whole point: the test runner will just pick up the deploy id from the deploy file.

>     * can we still set out own `deployment id` via env var?

Yes.

>     * what is the way to get same unique value through the run for
>       
>       * Prometheus - populated from Pod labels
>       * Elasticsearch - populated from _STATUS.log
 
The runid in the Pod labels should already be correct, and match what's in the status.
 
> I did a test run via GitHub Actions and there we still assign `runid` label/env manually via template, but
> 
>     * I can't see `deploymentid` label - this is used by Prometheus
> 
>     * In `09-56-44Z_ContinuousTestRun_STATUS.log` I see `"runid":"20240206-095227"`

Correct, I didn't change the external facing attribute names - sorry for not mentioning that sooner. They are still called runid both in the status log and the Prometheus labels.
 
>     * Actual `runid` set by GitHub Actions was `20240206-095044` - I will remove that when we will switch to `deployment id`

Yeah, for now what will happen is that if you don't set DEPLOYID then it will simply generate it automatically, using a very similar strategy to what you have in the github action.

File: Tests/DistTestCore/Logs/StatusLog.cs [hunk 1]
```diff
@@ -26,7 +28,7 @@ public void ConcludeTest(string resultStatus, TimeSpan testDuration, Dictionary<
         public void ConcludeTest(string resultStatus, string testDuration, Dictionary<string, string> data)
         {
             data.Add("timestamp", DateTime.UtcNow.ToString("o"));
-            data.Add("runid", NameUtils.GetRunId());
+            data.Add("runid", deployId);
```
Comment (benbierens) [inline Tests/DistTestCore/Logs/StatusLog.cs:31]: key is still "runid". Should it be "deployid"?

File: Tests/DistTestCore/TestLifecycle.cs [hunk 2]
```diff
@@ -76,7 +78,7 @@ public void OnContainersStopped(RunningContainers rc)
         public void OnContainerRecipeCreated(ContainerRecipe recipe)
         {
             recipe.PodLabels.Add("tests-type", TestsType);
-            recipe.PodLabels.Add("runid", NameUtils.GetRunId());
+            recipe.PodLabels.Add("runid", deployId);
```
Comment (benbierens) [inline Tests/DistTestCore/TestLifecycle.cs:81]: same here!

File: Tools/CodexNetDeployer/K8sHook.cs [hunk 3]
```diff
@@ -27,7 +29,7 @@ public void OnContainersStopped(RunningContainers rc)
         public void OnContainerRecipeCreated(ContainerRecipe recipe)
         {
             recipe.PodLabels.Add("tests-type", testsTypeLabel);
-            recipe.PodLabels.Add("runid", NameUtils.GetRunId());
+            recipe.PodLabels.Add("runid", deployId);
```
Comment (benbierens) [inline Tools/CodexNetDeployer/K8sHook.cs:32]: one more!

**Review by benbierens — APPROVED** [review]

File: Tests/DistTestCore/Logs/StatusLog.cs (hunk 1 above)
Comment (gmega) [reply Tests/DistTestCore/Logs/StatusLog.cs:31]: Yeah, I was delaying changing the actual label to sync up with @veaceslavdoina, maybe now is the time.

**Pull request merged**